### PR TITLE
Problem: Function RestoreClient is not aysnc (fix #244, #255)

### LINKED ISSUE #244: Problem: No events could be used to hide the QR Code

### LINKED ISSUE #255: Problem: Function RestoreClient is not aysnc

### PR DESCRIPTION
**Pull request by damoncro**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-com/play-unreal-plugin/blob/main/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest main? 
- [ ] Have you checked your code compiles in Unreal Engine 4 and 5?
- [ ] Have you checked your basic code formatting and style is fine? ([using this Linter plugin](https://ue4-style-guide.readthedocs.io/en/latest/gettingstarted.html))
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the plugin version number and documented your changes in the [CHANGELOG.md](https://github.com/crypto-com/play-unreal-plugin/blob/main/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-com/play-unreal-plugin/blob/main/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-com/play-unreal-plugin/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Review by tomtau — COMMENTED** [review]
why does RestoreClient need to be async? To my understanding, it'll just load the session file (which ok, it could be async, but in general, sync should be fast?) and open a websocket connection (but that happens in a background)... is it just for the case to have a notification back to the game if the connection failed to be established?

**Comment by damoncro** [comment]
> why does RestoreClient need to be async? To my understanding, it'll just load the session file (which ok, it could be async, but in general, sync should be fast?) and open a websocket connection (but that happens in a background)... is it just for the case to have a notification back to the game if the connection failed to be established?

Because it has noticeable blocking experience when I call it in game, it has to be async.

**Comment by damoncro** [comment]
> why does RestoreClient need to be async? To my understanding, it'll just load the session file (which ok, it could be async, but in general, sync should be fast?) and open a websocket connection (but that happens in a background)... is it just for the case to have a notification back to the game if the connection failed to be established?

Two events for notifications are actually different: One is `FOnNewSessionReady`, the other is `FOnRestoreSessionReady`. Those are very important to be used to hide the QR code.

**Review by tomtau — APPROVED** [review]
ok, I haven't tested it, but if laggy/blocking in 5.1, ok to make it async

File: Source/CronosPlayUnreal/Public/PlayCppSdkActor.h [hunk 1]
```diff
@@ -211,6 +231,14 @@ class CRONOSPLAYUNREAL_API APlayCppSdkActor : public AActor {
...
     FEnsureSessionDelegate OnEnsureSessionDelegate;
 
+    /**
+     * Used for `ConnectWalletConnect`
+     */
+    FString _description;
+    FString _url;
+    TArray<FString> _icon_urls;
+    FString _name;
+    int64 _chain_id;
```
[the remark this answers is not in the document]
Comment (damoncro) [reply Source/CronosPlayUnreal/Public/PlayCppSdkActor.h:241]: oh yes, should be in private.

[the remark this answers is not in the document]
Comment (damoncro) [reply Source/CronosPlayUnreal/Public/PlayCppSdkActor.h:241]: Checked, already in private section: https://github.com/cronos-labs/play-unreal-plugin/blob/8f26e0a748c85c25a8f6891b6620be5ad619931b/Source/CronosPlayUnreal/Public/PlayCppSdkActor.h#L208-L242

[the remark this answers is not in the document]
Comment (damoncro) [reply Source/CronosPlayUnreal/Public/PlayCppSdkActor.h:241]: These values are only used for function `ConnectWalletConnect` for users to configure when they call it: 
image, seems grouping them is inconvenient for users to set them (they need to make a struct first).

[the remark this answers is not in the document]
Comment (damoncro) [reply Source/CronosPlayUnreal/Public/PlayCppSdkActor.h:241]: They would be initialised when they call `ConnectWalletConnect`, empty by default except Connection Type.

[the remark this answers is not in the document]
Comment (damoncro) [reply Source/CronosPlayUnreal/Public/PlayCppSdkActor.h:241]: That's messy, so far I haven't found a better way to pass those values from ConnectWalletConnect to callback functions, except private values.

**Review by leejw51crypto — APPROVED** [review]
lgtm

**Pull request merged**